daemon/oci_linux.go

```diff
@@ -98,6 +98,30 @@ func setDevices(s *specs.Spec, c *container.Container) error {
 		for _, d := range hostDevices {
 			devs = append(devs, oci.Device(d))
 		}
+		for _, deviceMapping := range c.HostConfig.Devices {
+			if deviceMapping.PathOnHost == deviceMapping.PathInContainer {
+				logrus.Warnf("Specifying a device %s is redundant for privileged mode", deviceMapping.PathOnHost)
+				if deviceMapping.CgroupPermissions != "rwm" {
+					logrus.Warnf("Specify  permission %s to device %s   is redundant for privileged mode", deviceMapping.CgroupPermissions, deviceMapping.PathOnHost)
```

Comment:
Extra space after `Specify`
Also should be `Specifying`.
These could all use some extra context on the log messages... e.g. `logrus.WithField("container", c.ID)`